documentation/source/quickstart.rst

```diff
@@ -4,291 +4,189 @@
+    # Makefile
 
-.. code-block:: pycon
+    # defaults
+    SIM ?= icarus
+    TOPLEVEL_LANG ?= verilog
 
-    >>> # Read a value back from the DUT
-    >>> count = dut.counter.value
-    >>> print(count.binstr)
-    1X1010
-    >>> # Resolve the value to an integer (X or Z treated as 0)
-    >>> print(count.integer)
-    42
-    >>> # Show number of bits in a value
-    >>> print(count.n_bits)
-    6
+    VERILOG_SOURCES += $(PWD)/my_design.sv
+    # use VHDL_SOURCES for VHDL files
 
-We can also cast the signal handle directly to an integer:
+    # TOPLEVEL is the name of the toplevel module in your Verilog or VHDL file
+    TOPLEVEL = my_design
 
-.. code-block:: pycon
+    # MODULE is the basename of the Python test file
+    MODULE = test_my_design
 
-    >>> print(int(dut.counter))
-    42
+    # include cocotb's make rules to take care of the simulator setup
+    include $(shell cocotb-config --makefiles)/Makefile.sim
 
 
-Parallel and sequential execution
----------------------------------
+.. _quickstart_running_a_test:
 
-An :keyword:`await` will run an :keyword:`async` coroutine and wait for it to complete.
-The called coroutine "blocks" the execution of the current coroutine.
-Wrapping the call in :func:`~cocotb.fork` runs the coroutine concurrently, allowing the current coroutine to continue executing.
-At any time you can :keyword:`await` the result of the forked coroutine, which will block until the forked coroutine finishes.
+Running a Test
+==============
 
-The following example shows these in action:
+When you now type
 
-.. code-block:: python3
+.. code-block:: bash
 
-    # A coroutine
-    async def reset_dut(reset_n, duration_ns):
-        reset_n <= 0
-        await Timer(duration_ns, units='ns')
-        reset_n <= 1
-        reset_n._log.debug("Reset complete")
+   make
 
-    @cocotb.test()
-    async def parallel_example(dut):
-        reset_n = dut.reset
+Icarus Verilog will be used to simulate the Verilog implementation of the DUT because
+we defined these as the default values.
 
-        # Execution will block until reset_dut has completed
-        await reset_dut(reset_n, 500)
-        dut._log.debug("After reset")
+If you want to simulate the DUT with Siemens Questa instead,
+all you would need to change is the command line:
 
-        # Run reset_dut concurrently
-        reset_thread = cocotb.fork(reset_dut(reset_n, duration_ns=500))
+.. code-block:: bash
 
-        # This timer will complete before the timer in the concurrently executing "reset_thread"
-        await Timer(250, units='ns')
-        dut._log.debug("During reset (reset_n = %s)" % reset_n.value)
+    make SIM=questa
 
-        # Wait for the other thread to complete
-        await reset_thread
-        dut._log.debug("After reset")
 
-See :ref:`coroutines` for more examples of what can be done with coroutines.
+This concludes our quick introduction to cocotb.
+You can now look through our :ref:`tutorials` or check out the
```

Comment:
```suggestion
You can now look through our Tutorials or check out the
```

(The link doesn't work as there's no page, it instead points to he index page, which isn't really helpful.)